shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -1109,12 +1120,21 @@ public void onStop() {
     }
   }
 
+  @Override
+  public void onViewCreated(View view, Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+    view.getViewTreeObserver().addOnWindowFocusChangeListener(onWindowFocusChangeListener);
+  }
+
   @Override
   public void onDestroyView() {
     super.onDestroyView();
     if (stillAttachedForEvent("onDestroyView")) {
       delegate.onDestroyView();
     }
+    requireView()
+        .getViewTreeObserver()
+        .removeOnWindowFocusChangeListener(onWindowFocusChangeListener);
```

Comment:
You may wan to call this before calling `delegate.onDestroyView()`. I don't know if it matters though.

Reply:
Good point. Moved.